pkg/ext/extension_apiserver.go

```diff
@@ -317,3 +354,91 @@ func SetAggregationCheck(client wranglerapiregistrationv1.APIServiceClient, valu
 		return nil
 	})
 }
+
+func makePatchAndUpdateAPI(original, modified *apiregv1.APIService, apiservice wranglerapiregistrationv1.APIServiceController) ([]byte, error) {
+	originalJSON, err := json.Marshal(original)
+	if err != nil {
+		return nil, err
+	}
+	modifiedJSON, err := json.Marshal(modified)
+	if err != nil {
+		return nil, err
+	}
+	patch, err := jsonpatch.CreateMergePatch(originalJSON, modifiedJSON)
+	if err != nil {
+		return nil, err
+	}
+	if _, err := apiservice.Patch(APIServiceName, types.MergePatchType, patch); err != nil {
+		return patch, err
+	}
+	return patch, nil
+}
+
+func makePatchAndUpdateService(original, modified *corev1.Service, service wranglercorev1.ServiceController) ([]byte, error) {
+	originalJSON, err := json.Marshal(original)
+	if err != nil {
+		return nil, err
+	}
+	modifiedJSON, err := json.Marshal(modified)
+	if err != nil {
+		return nil, err
+	}
+	patch, err := jsonpatch.CreateMergePatch(originalJSON, modifiedJSON)
+	if err != nil {
+		return nil, err
+	}
+	var resources = ""
+
+	if _, err := service.Patch(Namespace, TargetServiceName, types.MergePatchType, patch, resources); err != nil {
+		return patch, err
+	}
+	return patch, nil
+}
+
+func DeleteLegacyServiceAndSecret(apiservice wranglerapiregistrationv1.APIServiceController, secrets wranglercorev1.SecretController) error {
```

Comment:
I might be missing something but I'm not seeing `DeleteLegacyServiceAndSecret` anywhere?